src/connectionController.ts

```diff
@@ -31,6 +24,24 @@ import { StorageVariables } from './storage';
 import type { StatusView } from './views';
 import type TelemetryService from './telemetry/telemetryService';
 import LINKS from './utils/links';
+import type {
+  ConnectionInfo as ConnectionInfoFromLegacyDS,
+  ConnectionOptions as ConnectionOptionsFromLegacyDS,
+} from 'mongodb-data-service-legacy';
+import {
+  getConnectionTitle,
+  extractSecrets,
+  mergeSecrets,
```

Comment:
These are now exported by `@mongodb-js/connection-storage`, and we may be able to use it, but that has a hard, non-optional/non-lazy dependency on `keytar`, so I don’t think we can use it here. Together with the item below, that made it seem very reasonable to just use a prior version of `mongodb-data-service` and take care of this issue when we actually do the VSCode connection improvements project.